### ISSUE TITLE
Support scaffold Spring Boot components

### ISSUE THREAD

**Issue by jdneo** [microsoft/vscode-java-dependency#767]
i.e. 

Automatically command palette/modal triggers and ask you for the entity name which will be applied in the following generated files inside they respective folders: Controllers/[EntityNameController.java] || BusinessLogic/Services/ServiceImpl [EntityNameServiceImpl.java] || BusinessLogic/Services/ServiceInterfaces[EntityNameService.java] || Models/Repositories/[EntityNameDao.java] || Models/Entities/[EntityName.java]

Maybe need some integration tasks to do with the spring extension.

**Comment by BizarreBoy** [comment]
Hi! 

Following with the scaffold idea, I would like to attach some screenshots.

In the first picture I created by hand a sample bunch of directories with one file inside each one related to the function they were created for. (Imagine it has been generated with a context menu item named for example "Generate Initial Java Scaffold API" supposing I right-clicked in the named "rest" folder of the following image example):

![image](https://github.com/microsoft/vscode-java-dependency/assets/67835936/2d9078a6-8d39-421e-8149-51aa8b9235a2)

The feature should open a command palette modal and ask to the user what will be the entity name to be applied in any related file generated, for example: SampleNameEntity, SampleNameController, SampleNameService, SampleNameDTO,...

![ss02](https://github.com/microsoft/vscode-java-dependency/assets/67835936/a7794bf9-91cf-4c47-8ce5-f5d5e7f75639)

The initial code by default could be empty methods with the params names replaced by the required entity name from before:

NOTE: In the following pictures you will see default templates generated by the plugin Spring Code Generator developed by the author Sonal Sithara. His plugin saves a lot of time typing general methods and let user focus on the logic development and not in the code structure.

CONTROLLER:

![image](https://github.com/microsoft/vscode-java-dependency/assets/67835936/6cc77310-eec2-400a-b258-ed4f635af9e7)

SERVICE IMPLEMENTATION:

![image](https://github.com/microsoft/vscode-java-dependency/assets/67835936/634d809d-1a92-45ad-8fe3-253180e66c38)

SERVICE INTERFACE:

![image](https://github.com/microsoft/vscode-java-dependency/assets/67835936/dda9ec8c-69cd-42e7-ae75-3102ad788b73)

REPOSITORY:

![image](https://github.com/microsoft/vscode-java-dependency/assets/67835936/a68ba9fd-f9a4-49ab-aead-8591b72e01d4)

ENTITY:

![image](https://github.com/microsoft/vscode-java-dependency/assets/67835936/7c4f1519-fb64-446d-a53a-2053f76b57c4)

RESUME: The idea would be a context menu option which open a command palette modal to insert the shared EntityName. Then the tool will create the required mkdirs and generate inside of each one the file templates according to each folder functionality. 

BONUS: Would be great a second modal question for the user before to start generating the scaffold (optional) which ask for Lombok Dependency integration and generates the initial entity template according to this choice. 
I mean: [NO] getters and setters generated typed in the code || [YES] just lombok annotation like @Data/@Getter/@Setter/@AllArgsConstructor.

**Comment by jdneo** [comment]
Thank you @BizarreBoy for providing those detailed information.

BTW, just for asking, do other Java IDEs/tools provide feature similar to this one? If there is, we can take a look and make it as a reference.

**Comment by jdneo** [comment]
Thank you. The idea is clear. But we need to think more about how to make the generated structure be suitable to users project as much as possible. (or, make it be easy to configure/customize)

**Comment by BizarreBoy** [comment]
I agree, in my opinion we could take the angular schematics plugin as reference. 
With just one click and a command palette question it quickly creates the starter scaffold of a new component with a few files related with each other.

**Comment by jdneo** [comment]
There are two options in my mind now:

1. You mentioned the extension `Spring Code Generator`, which I took a very quick glance on that and looks very nice. We can consider contributing to that extension and let it hook those command to the Java Project explorer, then as long as user installs the Project Manager extension and Spring Code Generator, they can see those actions in the context menu.

2. Leverage Copilot to scaffold the template. I just tried it in Copilot Chat and the result looks promising 😁
![image](https://github.com/microsoft/vscode-java-dependency/assets/6193897/f60c38be-b24a-4e2f-9bfb-fff22b489aff)

**Comment by BizarreBoy** [comment]
Wow!!! Amazing job by copilot! 

And yes, Spring Code Generator works similar, the only thing it doesn't create is the folder structure and only generates files one by one which is comfortable in most scenarios but a bit annoying when user has to create a full APIREST from zero.

This is the reason why I focus on Angular Schematics reference. But it's true that Java Scaffold should work different because Springboot users need the main scaffold just one by project/microservice: Controllers | BussinesLogic [ Services | Interfaces ] | Models [ Entities | DTO | DAO(Repositories) ].

Once the main scaffold is created, the tool should work exactly as Spring Code Generator plugin and add the other templates one by one as the user needs. 

Let me know your opinion about the functionality/workflow I described you.
